### ISSUE TITLE
Can not build v1.0.0

### ISSUE THREAD

**Issue by Ingvord** [pni-libraries/libpnicore#39]
Dear Maintainers,

I was trying and failed to build pnicore-v1.0.0.

Compilation failure:

```
Scanning dependencies of target array_objects
[ 64%] Building CXX object src/arrays/CMakeFiles/array_objects.dir/array_selection.cpp.o
In file included from /libpnicore/src/arrays/array_selection.cpp:25:0:
/libpnicore/./include/pni/core/arrays/array_selection.hpp: In member function 'size_t pni::core::array_selection::size() const':
/libpnicore/./include/pni/core/arrays/array_selection.hpp:298:24: error: 'accumulate' is not a member of 'std'
                 return std::accumulate(_oshape.begin(),_oshape.end(),
                        ^~~
make[2]: *** [src/arrays/CMakeFiles/array_objects.dir/array_selection.cpp.o] Error 1
src/arrays/CMakeFiles/array_objects.dir/build.make:62: recipe for target 'src/arrays/CMakeFiles/array_objects.dir/array_selection.cpp.o' failed
CMakeFiles/Makefile2:1423: recipe for target 'src/arrays/CMakeFiles/array_objects.dir/all' failed
make[1]: *** [src/arrays/CMakeFiles/array_objects.dir/all] Error 2
make: *** [all] Error 2
Makefile:138: recipe for target 'all' failed
The command '/bin/sh -c make install' returned a non-zero code: 2
```

Corresponding Docker file:

```docker
FROM debian:stretch AS pnicore

RUN apt-get update && apt-get install -y \
  apt-utils                              \
  build-essential                        \
  cmake                                  \
  curl                                   \
  git                                    \
  lsb-release                            \
  libboost-system-dev                    \
  libboost-regex-dev                     \
  libboost-program-options-dev           \
  libcppunit-dev

RUN git clone https://github.com/pni-libraries/libpnicore.git

WORKDIR libpnicore

RUN git checkout v1.0.0 && mkdir build

WORKDIR build

RUN cmake -DCMAKE_INSTALL_PREFIX=/opt/pni ..

RUN make install
```

**Comment by jkotan** [comment]
Hi Igor,

this bug was fixed in 1.1.0.  Do you really need 1.0.0?

**Comment by Ingvord** [comment]
@jkotan thanks for the response. I need to check if it is compatible with the rest of the system we are using

**Comment by Ingvord** [comment]
Fixed in  e30f92d
